### ISSUE TITLE
Bookmarks config file

### ISSUE THREAD

**Issue by tsujan** [lxqt/libfm-qt#1009]
For bookmarks we use `~/.config/gtk-3.0/bookmarks` to have compatibility with GTK3 apps. We also check the legacy file of GTK2 (why?).

I don't know about GTK4, but I think following GTK isn't a good idea after GTK3. Instead, I propose that we forget about GTK ≥ 4 and take a more self-consistent approach, by using a file of our own (perhaps `~/.local/share/libfm-qt/bookmarks`), with `~/.config/gtk-3.0/bookmarks` only for backward compatibility.

**Comment by stefonarch** [comment]
>We also check the legacy file of GTK2 (why?).

Because of pcmanfm?  Afaik only arch has pcmanfm-gtk3.

Atm bookmarks are shared with thunar, nautilus and others probably. So existing bookmarks will be ported to a new config file? GTK4 should provide backwards compatibility too, so the current approach won't be unusable soon IMO.

**Comment by tsujan** [comment]
> So existing bookmarks will be ported to a new config file?

Yes, it'll be backward compatible (I already made a patch): After the first run of any libfm-qt based app, libfm-qt will make its separate bookmarks file by using the GTK3 file (if any) and won't share any change to it it with GTK afterward. Some users might consider it as a good thing because their GTK bookmarks will be kept untouched.

> GTK4 should provide backwards compatibility too

Do you really believe backward compatibility from GNOME?  ;)

BTW, do you know how GTK4 manages bookmarks? I installed `gtk4-demos` and found a way of calling open-file dialog from `gtk4-widget-factory` but, to my surprise, our LXQt file dialog was launched instead!

**Comment by tsujan** [comment]
> Maybe the lxqt portal?

Yes. I just wanted to see how a purely GTK4 file dialog managed bookmarks. But it isn't important.

> Let's make a PR...

Will do it soon.
